[Event Hubs] Processor Improvements (#29615)
Azure/azure-sdk-for-net: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md

### Features Added

> - Added a warning surfaced to the event processor error handler signaling that an unhandled exception propagated from the event processing handler causing partition processing to fault and restart.

Reviewer: ```suggestion
- The event processor error handler will now raise warning when an unhandled exception propagated from the event processing handler causing partition processing to fault and restart.
```
